restapi/src/main/java/io/stargate/web/docsapi/service/query/ExpressionParser.java

```diff
@@ -85,26 +96,99 @@ public ExpressionParser(ConditionParser predicateProvider) {
    */
   public List<Expression<FilterExpression>> parse(
       List<PathSegment> prependedPath, JsonNode filterJson, boolean numericBooleans) {
+    return parse(
+        prependedPath, Collections.singletonList(filterJson), numericBooleans, new MutableInt(0));
+  }
+
+  /**
+   * Parses set of json nodes and returns back the list of the expressions found. Note that
+   * expression in the list contain only first level conditions in each node.
+   *
+   * @param prependedPath Given collection or document path segments
+   * @param nodes JSON nodes to resolve
+   * @param numericBooleans If number booleans should be used when creating conditions.
+   * @param nextIndex index counter for the expressions
+   * @return List of all expressions
+   */
+  private List<Expression<FilterExpression>> parse(
```

Comment:
Would it make sense to return a single `Expression` to encapsulate all boolean operators? With current method signature I guess the individuals expressions in the list are implicitly joined by an `AND`, right? I think it would be nice to make that explicit.

Reply:
This is a private method.. It's clearly specified in the JavaDocs what it does. There is another method for wrapping. Plus, it can not be done due to the recursion!

Reply:
Re: `private` - do we not have the same return type in the `public` entry point?

Reply:
Re: recursion - agreed. I should have made the comment on the `public` method, sorry about the confusion.

Reply:
The public method that returns the `List` is not used actually, I can make it private and make all tests go through that only public method?

Reply:
Yes, please. I think it would be more consistent to test through the public method that returns an `Expression`. Let's keep the internal method `private`.

Reply:
Done.